lib/system/connection_probe.rb

```diff
@@ -26,17 +21,10 @@ def ready?
         end
 
         def sql_for_readiness
-          if oracle?
-            <<~SQL
-              SELECT 1 FROM v$database WHERE cdb = 'NO' AND open_mode = 'READ WRITE'
-              UNION ALL
-              SELECT 1 FROM v$pdbs WHERE name COLLATE BINARY_CI = '#{connection_config[:database]}' AND open_mode = 'READ WRITE'
-            SQL
-          elsif mysql?
-            'SELECT 1'
-          else
-            'SELECT 1'
-          end
+          # We include "dummy" in the Oracle query to skip FAST DUAL and hit the database.
+          return 'SELECT 1, dummy FROM DUAL' if oracle?
```

Comment:
This is rather interesting and cool. Have you tried running it just after oracle container started to see if it will fail until DB was really available?

Reply:
I tried and it failed as expected.

Reply:
Great, that's much simpler than previous version.